sources/packages/backend/apps/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -212,6 +235,47 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
     return found.length === locations.length;
   }
 
+  /**
+   * Check if location code is already registered for the institution.
+   * @param locationCode location code.
+   * @param locationId location id in case it is an update of the location.
+   * @param institutionId institution id in case it is a new location.
+   * @returns true in case there is already a location code for the institution.
+   */
+  async hasLocationCodeForInstitution(
+    locationCode: string,
+    locationId?: number,
+    institutionId?: number,
+  ): Promise<boolean> {
+    const institutionIdQueryBuilder = await this.repo
+      .createQueryBuilder("sublocation")
+      .select("sublocation.institution.id")
+      .where("sublocation.id = :locationId", { locationId });
+
+    const queryBuilder = this.repo
+      .createQueryBuilder("location")
+      .select("1")
+      .where("location.institutionCode = :locationCode", { locationCode });
+
+    if (locationId) {
+      queryBuilder
+        .andWhere(
+          "location.institution.id = (" +
+            institutionIdQueryBuilder.getQuery() +
+            ")",
+        )
+        .setParameters(institutionIdQueryBuilder.getParameters())
+        .andWhere("location.id != :locationId", { locationId });
+    } else {
```

Comment:
The parameter `institutionId` must be tested also with an `if` to ensure that it was provided.
In case no optional parameters were provided an exception can be thrown, like `thorw new Error("At least one optional parameter is required.")`.

Reply:
ok. Done, Thanks!